### PR TITLE
Tests for set and option are not comprehensive

### PR DESCRIPTION
**Pull request by Milly**
- Moved tests for `:set` to test_set.vim.
- Added comprehensive test patterns.
- Split up long errors test into separate tests for each options.
- Reordering option tests alphabetically.

### CODE REVIEW

**Comment by chrisbra** [comment]
Thanks for adding new tests, but unfortunately, there are a few failures:
```
	From test_options.vim:
	Found errors in Test_opt_writedelay():
	command line..script D:/a/vim/vim/src/testdir/runtest.vim[607]..function RunTheTest[57]..Test_opt_writedelay line 13: Expected range 0.3 - 999.0, but got 0.002934
	Found errors in Test_set_completion_string_values():
	command line..script D:/a/vim/vim/src/testdir/runtest.vim[607]..function RunTheTest[57]..Test_set_completion_string_values line 122: Expected 'tagfile' but got 'fuzzy'
```

The second one seems to be caused by `getcompletion()` returning the results in different orders on Windows vs. Unix (at least parly). 
The first one however is  a bit puzzling, may be that is flaky. 

And then there is this failure on MacOS:
```
	From test_alot.vim:
	Found errors in Test_reltime():
	Run 1, 03:03:14 - 03:03:14:
	command line..script /Users/runner/work/vim/vim/src/testdir/runtest.vim[607]..function RunTheTest[57]..Test_reltime line 6: Expected True but got 0
	command line..script /Users/runner/work/vim/vim/src/testdir/runtest.vim[607]..function RunTheTest[57]..Test_reltime line 8: Expected True but got 0
	command line..script /Users/runner/work/vim/vim/src/testdir/runtest.vim[607]..function RunTheTest[57]..Test_reltime line 16: Expected True but got 0
	command line..script /Users/runner/work/vim/vim/src/testdir/runtest.vim[607]..function RunTheTest[57]..Test_reltime line 18: Expected True but got 0
	Run 2, 03:03:16 - 03:03:16:
	command line..script /Users/runner/work/vim/vim/src/testdir/runtest.vim[642]..function RunTheTest[57]..Test_reltime line 6: Expected True but got 0
	command line..script /Users/runner/work/vim/vim/src/testdir/runtest.vim[642]..function RunTheTest[57]..Test_reltime line 8: Expected True but got 0
	command line..script /Users/runner/work/vim/vim/src/testdir/runtest.vim[642]..function RunTheTest[57]..Test_reltime line 16: Expected True but got 0
	command line..script /Users/runner/work/vim/vim/src/testdir/runtest.vim[642]..function RunTheTest[57]..Test_reltime line 18: Expected True but got 0
	Run 3, 03:03:18 - 03:03:18:
	command line..script /Users/runner/work/vim/vim/src/testdir/runtest.vim[642]..function RunTheTest[57]..Test_reltime line 6: Expected True but got 0
	command line..script /Users/runner/work/vim/vim/src/testdir/runtest.vim[642]..function RunTheTest[57]..Test_reltime line 8: Expected True but got 0
	command line..script /Users/runner/work/vim/vim/src/testdir/runtest.vim[642]..function RunTheTest[57]..Test_reltime line 16: Expected True but got 0
	command line..script /Users/runner/work/vim/vim/src/testdir/runtest.vim[642]..function RunTheTest[57]..Test_reltime line 18: Expected True but got 0
	Flaky test failed too often, giving up
```
It seems reltime() doesn't work properly there? Perhaps also flaky?

**Comment by zeertzjq** [comment]
> Reordering option tests alphabetically.

I think this makes the changes hard to follow. Maybe it should be moved to a separate PR, if it should be done at all.

**Comment by chrisbra** [comment]
Thanks for reminding me. I wanted to mention that one as well. Please don't re-order the tests, especially if you are also renaming the tests. Tests are executed in sorted order anyhow and is we add more tests, they typically get added at the end of the file (so sorting would be broken again).

**Comment by Milly** [comment]
> Please don't re-order the tests
OK, I'm working.

**Comment by chrisbra** [comment]
yeah, let's keep it in test_options. I don't think it matters to much if a specific test is in `test_options` or in `test_set`. thanks

**Comment by Milly** [comment]
I re-PR'd in #15856.
